### PR TITLE
docs+tests(bias): pin set/change-by-statistic semantics for bridged models

### LINKED ISSUE #5927: set-by-statistic double-counts the model-level analytical contribution for bridged models
## Summary

For a model that adds an analytical contribution at the **model** layer (SeZM + `InnerPotential`/ZBL bridging), `bias_adjust_mode="set-by-statistic"` fits the atomic output bias to the raw labels. Because the analytical term is then added on top of the atomic output at inference, the mean analytical contribution is counted twice.

`change-by-statistic` was recently corrected for this in #5910, which routes it through a complete model-level predictor. `set-by-statistic` was left as-is, so the two modes now disagree about what the bias represents for bridged models.

## Mechanism

`compute_output_stats` chooses what gets fitted based on whether a predictor is supplied:

https://github.com/deepmodeling/deepmd-kit/blob/799252d40da5e4d9a9f48ef1444cd4936964a4ab/deepmd/pt/utils/stat.py#L678-L687

With `model_forward=None` the fit target is `merged_output`, i.e. the raw labels. `set-by-statistic` is exactly the branch that passes no predictor:

https://github.com/deepmodeling/deepmd-kit/blob/799252d40da5e4d9a9f48ef1444cd4936964a4ab/deepmd/pt/model/atomic_model/base_atomic_model.py#L630-L645

For a bridged model the label is the true total energy, analytical term included. So the per-type bias absorbs the mean analytical contribution, and the model then adds that contribution again during the forward pass.

## Why the #5910 fix does not cover it

#5910 added a model-level `change_out_bias` override that short-circuits `change-by-statistic` through `predict_atomic_outputs_for_stat`, and delegates every other mode to the atomic model unchanged. Separately, the initial statistics path never reaches the model layer at all: `compute_or_load_out_stat` is an atomic-model method calling the atomic `change_out_bias` directly, so a model-level override cannot intercept it.

https://github.com/deepmodeling/deepmd-kit/blob/799252d40da5e4d9a9f48ef1444cd4936964a4ab/deepmd/pt/model/atomic_model/base_atomic_model.py#L565-L569

## Where this is reachable

Three call sites use `set-by-statistic`:

1. `compute_or_load_out_stat` — the initial output-statistics computation, run at the start of **every** training run (link above).
2. `dp change-bias --mode set` — https://github.com/deepmodeling/deepmd-kit/blob/799252d40da5e4d9a9f48ef1444cd4936964a4ab/deepmd/pt/entrypoints/main.py#L608-L612
3. The finetune path in `deepmd/pt/train/training.py` around L955.

Case 1 is self-correcting in the sense that training can learn to compensate a poor initial bias, so the practical cost there is a worse starting point rather than a permanent error. Cases 2 and 3 apply the shift to an already-trained model, where nothing corrects it.

## What I have not verified

- The **magnitude** of the effect. It is the mean analytical contribution over the calibration set, which for a short-range repulsive term at equilibrium geometries may be small. Somebody should measure it before deciding how urgent this is.
- Whether `DPZBLModel` (the separate `DPZBLLinearEnergyAtomicModel` blending implementation used by dpmodel/pt_expt) is affected. It composes differently — the blend happens inside the atomic model, and it is a smooth switch rather than a plain sum — so the analysis above does not carry over directly.
- Whether the current behaviour is in fact intended for some workflow I have not considered.

This is **not** a regression from #5910; the same behaviour is present before it. It surfaced while reviewing that PR, because fixing one mode made the asymmetry between the two visible.

## Possible direction

Neither existing mode expresses the right thing. Passing the model-level predictor to `set-by-statistic` would just turn it into `change-by-statistic`. What the bridged case needs is to subtract only the analytical contribution from the labels and then fit absolutely — a third behaviour. A narrower alternative is to reject `set-by-statistic` for models carrying a model-level analytical term, so the wrong number is never silently produced.

Whichever direction is taken, a regression test in the shape of `test_change_out_bias_is_invariant_for_self_labels` from #5910 would pin it: label the data with the model's own output, apply the mode, and assert the resulting predictions still reproduce the labels.

### PR DESCRIPTION
**Pull request by wanghan-iapcm**
Close #5927 — resolved **as designed**, with documentation and semantic pin tests instead of a behavior change.

## The definitions (verified against the code)

The model energy decomposes as `E = E_model + E_bias`, where `E_model` is everything the model computes (learned network, analytical ZBL term, or a `linear_ener` combination) and `E_bias` is the per-type output bias.

- **`set-by-statistic`** assigns `E_bias` directly: user values, or the per-type least-squares statistic of the **raw labels**. It never calls a model forward (`compute_output_stats` with `model_forward=None`, stored with `add=False`) — it is independent of `E_model` **by definition**, ignoring a trained network and the analytical ZBL term alike.
- **`change-by-statistic`** fits the residual of the labels against the **complete model prediction** (the bridged predictor since #5910) and adds the delta to the existing bias.

Under these definitions the "double count" described in the issue is not a bridging bug: `set` uniformly ignores all of `E_model` for every model kind. A bridged model after `set` carries no compensation for the mean ZBL contribution — exactly as a trained plain model after `set` carries no compensation for its network output. The two modes "disagree" because they are defined to answer different questions; a self-consistent calibration is `change`'s job.

Verified conformance of the `linear_ener` composition path in both backends: children compute no output statistics (`compute_or_load_out_stat=False`); the composition level performs one `set` fit on the raw labels. Also verified: the ZBL term contributes exactly zero for isolated atoms, so its own statistics are trivially `bias = 0`.

## Changes

- `doc/model/change-bias.md`: precise definitions of the two modes, and the guidance that calibrating a bridged (or any nonzero-`E_model`) model self-consistently requires `change-by-statistic`.
- `doc/model/dpa4.md`: note in the ZBL section.
- Semantic pin tests (pt `SeZMModel` + dpmodel `LinearEnergyAtomicModel` composition): `set-by-statistic` equals the raw-label least-squares fit exactly — guarding against a future "fix" that would subtract the analytical term and silently create a third, model-dependent mode.

## Known limitations

- No behavior change anywhere; the pin tests cover `model.change_out_bias` (the `dp change-bias --mode set` and finetune routes) and the dpmodel composition out-stat; the pt training-init chain was verified in-session and funnels into the same pinned branch.
- The spin variants share the machinery but have no dedicated pin.
- Fact worth knowing when reading the docs: `InnerPotential` adds the full ZBL over the whole cutoff (not only below `bridging_r_outer`), so the label-side ZBL at equilibrium geometries is small but not strictly zero; the docs state the offset plainly.

## Summary by CodeRabbit

* **Documentation**
  * Clarified how `set` and `change` statistic modes handle model energy, analytical contributions, labels, residuals, and existing bias.
  * Documented the interaction between output-bias calibration and ZBL bridging, including guidance for self-consistent calibration.

* **Tests**
  * Added regression coverage confirming that `set-by-statistic` calibration uses raw energy labels independently of learned and ZBL model contributions.
  * Verified calibration correctly replaces seeded bias and remains consistent when repeated.

### CODE REVIEW

**Review by njzjz — COMMENTED** [review]
@njzjz-bot has commented above.

File: doc/model/change-bias.md [hunk 1]
```diff
@@ -9,6 +9,32 @@ There are several scenarios where one might want to adjust the output bias after
...
+per-type output bias.
+
+- **`set` (`set-by-statistic`)** assigns `E_bias` directly: either the
+  user-given values (`-b`), or the per-type least-squares statistic of the
+  **raw data labels**. It is independent of `E_model` by definition — it
+  ignores a trained network, and it equally ignores an analytical
+  contribution such as the ZBL term of a bridged model. The result is
+  reproducible and idempotent for a given dataset, but it contains **no
+  compensation for `E_model`**: after `set`, predictions on the calibration
+  data are offset by the data mean of `E_model`.
```
[the remark this answers is not in the document]
Comment (wanghan-iapcm) [reply doc/model/change-bias.md]: Fixed in https://github.com/deepmodeling/deepmd-kit/commit/a6a2bf690c67af49dc2ff0d708d55943b558a37c: the "offset by the data mean of `E_model`" claim is replaced by the accurate statement — after `set`, the remaining error on the calibration data is the configuration-dependent `E_model` itself plus any residual of the raw-label least-squares fit.

File: doc/model/change-bias.md [hunk 2]
```diff
@@ -9,6 +9,32 @@ There are several scenarios where one might want to adjust the output bias after
...
+- **`change` (`change-by-statistic`)** assigns `E_bias` from the residual:
+  the per-type statistic of the labels **minus the complete model
+  prediction** (including any analytical bridging term), added to the
+  existing bias. Use this mode for a self-consistent calibration of a
+  trained (or bridged) model.
+
+For a bridged model — or any model whose `E_model` is significantly nonzero
+on the calibration data — `set` therefore yields a bias that double-counts
+nothing and compensates nothing; if you want the model predictions to match
+the calibration labels, use `change`.
```
[the remark this answers is not in the document]
Comment (wanghan-iapcm) [reply doc/model/change-bias.md]: Fixed in https://github.com/deepmodeling/deepmd-kit/commit/a6a2bf690c67af49dc2ff0d708d55943b558a37c using your suggested text verbatim: the paragraph now describes that the raw-label fit can absorb the composition-correlated component of `E_model` into `E_bias` (which the forward pass then adds again), instead of claiming `double-counts nothing`, and keeps the guidance to use `change`.

File: source/tests/pt/model/test_sezm_model.py [hunk 3]
```diff
@@ -2201,6 +2201,56 @@ def test_change_out_bias_is_invariant_for_self_labels(self) -> None:
...
+                }
+            )
+            labels.append(label)
+            counts_rows.append(counts)
+        model.change_out_bias(samples, bias_adjust_mode="set-by-statistic")
+        bias = model.get_out_bias().detach().cpu().numpy().reshape(-1)[:2]
+        raw_fit = np.linalg.solve(
+            np.array(counts_rows, dtype=np.float64), np.array(labels)
+        )
+        np.testing.assert_allclose(bias, raw_fit, atol=1.0e-8)
```
[the remark this answers is not in the document]
Comment (wanghan-iapcm) [reply source/tests/pt/model/test_sezm_model.py:2255]: Fixed in https://github.com/deepmodeling/deepmd-kit/commit/a6a2bf690c67af49dc2ff0d708d55943b558a37c as suggested: the pt test now seeds a nonzero `out_bias` via `set_out_bias(ones_like(...))` before the call (an accidental `add=True` implementation would shift the result and fail), and repeats the call to pin idempotence. The dpmodel test mirrors both pins on its separate bias storage (`atomic_model.out_bias` seeding + repeated `compute_or_load_out_stat`). Both pass.

**Review by njzjz — APPROVED** [review]
